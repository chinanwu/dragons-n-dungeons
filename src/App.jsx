import React from 'react';
import Tabs from '@chinanwu/tabular';
import Tab from '@chinanwu/tabular';

const App = () => (
  <div>
    <Tabs>
      <Tab name="main">
        <p>I am main page</p>
      </Tab>
      <Tab name="inventory">
        <p>Inventory</p>
      </Tab>
    </Tabs>
  </div>
);

export default App;
