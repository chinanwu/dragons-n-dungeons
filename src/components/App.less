@import 'styling/colours.less';

* {
  font-family: 'Rubik', sans-serif;
}

html,
body {
  height: 100%;
  background-color: @main-0;
}

body {
  margin: 0;
}

#root {
  height: 100%;
}
