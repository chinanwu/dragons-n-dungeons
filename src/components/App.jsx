import React from 'react';
import { Provider } from 'react-redux';
import { store } from '../store.js';

import TabFooter from './tabs/TabFooter.jsx';
import TabHeader from './tabs/TabHeader.jsx';
import Tab from './tabs/Tab.jsx';
import Tabs from './tabs/Tabs.jsx';
import Logo from './general/Logo.jsx';
import Dropdown from './general/Dropdown.jsx';
import Main from './main/Main.jsx';

import './App.less';

export const App = () => (
  <Provider store={store}>
    <Tabs>
      <TabHeader>
        <Logo />
      </TabHeader>

      <Tab name="Main ">
        <Main />
      </Tab>
      <Tab name="Inventory">Inventory</Tab>
      <Tab name="Spells">Spells</Tab>

      <TabFooter>
        <Dropdown
          id="themeDropdown"
          name="Theme"
          options={['Light', 'Dark', 'Halloween']}
        />
      </TabFooter>
    </Tabs>
  </Provider>
);
export default App;
