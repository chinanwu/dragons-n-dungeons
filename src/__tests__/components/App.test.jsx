import { shallow } from 'enzyme';
import React from 'react';

import { App } from '../../components/App.jsx';

jest.unmock('../../components/App.jsx');

describe('App component', () => {
  describe('rendering', () => {
    it('renders App', () => {
      const wrapper = shallow(<App />);
      expect(wrapper.getElement()).toMatchSnapshot();
    });
  });
});
