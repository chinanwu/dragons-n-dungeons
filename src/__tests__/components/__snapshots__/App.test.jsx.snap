// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component rendering renders App 1`] = `
<Provider>
  <Memo(Connect(Tabs))>
    <TabHeader>
      <Memo(Connect(Logo)) />
    </TabHeader>
    <Tab
      disabled={false}
      name="Main "
    >
      <Main />
    </Tab>
    <Tab
      disabled={false}
      name="Inventory"
    >
      Inventory
    </Tab>
    <Tab
      disabled={false}
      name="Spells"
    >
      Spells
    </Tab>
    <TabFooter>
      <Memo(Connect(Dropdown))
        id="themeDropdown"
        name="Theme"
        options={
          Array [
            "Light",
            "Dark",
            "Halloween",
          ]
        }
      />
    </TabFooter>
  </Memo(Connect(Tabs))>
</Provider>
`;
